src/utils/request.ts

```diff
@@ -49,24 +49,44 @@ export function request(options: IRequestOptions): Promise<IRequestContext> {
 			opts.auth = options.user + ':' + options.password;
 		}
 
-		req = protocol.request(opts, (res: http.ClientResponse) => {
-			const followRedirects = isNumber(options.followRedirects) ? options.followRedirects : 3;
+		if (protocol === "https")
```

Comment:
I agree. But the TypeScript compiler gave some error "Type lacks a call signature" which I wasn't able to figure out how to fix. I hope you can fix this by a commit.